omnibus/files/private-chef-ctl-commands/reindex.rb

```diff
@@ -1,13 +1,143 @@
-# Copyright (c) 2013 Opscode, Inc.
-# All Rights Reserved
+#
+# Copyright 2015 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'optparse'
+require 'chef/config'
+require 'chef/rest'
+require 'chef/org'
+require 'redis'
+
+def all_orgs
+  Chef::Config.from_file("/etc/opscode/pivotal.rb")
+  Chef::Org.list.keys
+end
+
+def expander_queue_size
+  output = `/opt/opscode/embedded/service/rabbitmq/sbin/rabbitmqctl list_queues -p /chef | awk '{sum += $2} END {print sum}'`
+  status = $?
+  if !status.success?
+    $stderr.puts "Failed to get queue size!"
+    exit 1
+  else
+    output.to_i
+  end
+end
+
+def wait_for_empty_queue
+  loop do
+    size = expander_queue_size
+    if size == 0
+      break
+    else
+      puts "\t#{size} objects remaining in queue"
+      sleep 1
+    end
+  end
+end
+
+def enqueue_data_for_org(org)
+  status = Dir.chdir(File.join(base_path, "embedded", "service", "opscode-erchef", "bin")) do
+    run_command("./reindex-opc-organization complete #{org} >/dev/null")
+  end
+  if !status.success?
+    $stderr.puts "Failed to enqueue data for #{org}!"
+  end
+end
+
+def redis
+  @redis ||= begin
+                 vip = running_config["private_chef"]["redis_lb"]["vip"]
+                 port = running_config["private_chef"]["redis_lb"]["port"]
+                 password = running_config["private_chef"]["redis_lb"]["password"]
+                 Redis.new(port: port, ip: vip, password: password)
+             end
+end
+
+def disable_api
+  puts "- Disabling the Chef API."
+  redis.hset("dl_default", "503_mode", true)
+end
+
+def enable_api
+  puts "- Re-enabling the Chef API"
+  redis.hdel("dl_default", "503_mode")
```

Comment:
Do we want to delete this instead of setting it to true? I'm sure it does the same thing, I just didn't know if we had a convention.